### PR TITLE
reference using setuptools instead of pip internal

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
Issue: #113 

Reference function using setuptools instead of pip internal function. Reference: https://github.com/pypa/pip/issues/5243

### CODE REVIEW

File: piplicenses.py [hunk 1]
```diff
@@ -37,31 +37,7 @@
...
-            env = get_default_environment()
-            dists = env.iter_installed_distributions(
-                local_only=True,
-                skip=stdlib_pkgs,
-                include_editables=True,
-                editables_only=False,
-                user_only=False,
-            )
-            return [dist._dist for dist in dists]
+import pkg_resources
```
Comment (raimon49) [inline piplicenses.py:40]: I understood that it is better not to import functions from `pip._internal`. However, this ported code seems to have the advantage of working in many user environments.

Will `import pkg_resources` succeed in all user environments? For example, it seems to exist in `pip._vendor.pkg_resources`.

**Comment by jpmn** [comment]
Any hope in seeing a fix for this issue merged in? Using `pip=21.3.1`, it throws an exception:

```
File ".venv\Scripts\pip-licenses.exe\__main__.py", line 4, in <module>
File ".venv\lib\site-packages\piplicenses.py", line 44, in <module>
  from pip import get_installed_distributions
ImportError: cannot import name 'get_installed_distributions' from 'pip' (.venv\lib\site-packages\pip\__init__.py)
```

**Comment by raimon49** [comment]
@jpmn I haven't decided if I want to merge this PR or not.

Please install the latest version of pip-licenses in your environment and see if it solves the error.

```bash
# Update pip-licenses version to 3.5.3 or later
$ pip install -U pip-licenses
```

**Comment by jpmn** [comment]
I am using the latest version of `pip-licenses`, but I had to freeze my `pip` version until there is a fix, which isn't good. I might consider dropping the library in the medium-term if it is blocking my upgrade of `pip`.

Don't you experience the same behavior with more recent versions of `pip`?

**Comment by worroc** [comment]
This fix is required for many users, it is not acceptable to pin the pip version for a such long time. Is there road map to support pip >= 22?

**Comment by raimon49** [comment]
@worroc Who are "many users"? If this is about you, we need information about your environment.

At least in my environment it is working with pip 22.x.
```bash
$ pip-licenses --with-system
 Name               Version   License
 PTable             0.9.2     BSD License
 botocore           1.21.53   Apache Software License
 jmespath           0.10.0    MIT License
 pip                22.2.2    MIT License
 pip-licenses       3.5.3     MIT License
```

**Comment by raimon49** [comment]
This proposal was superseded by another excellent proposal that uses importlib\_metadata.

**Pull request closed without merging**